README.md

```diff
@@ -1,15 +1,126 @@
+| web (u)   | Starts a web server for the current Holochain app                   |
+
+## How To Get Started Building An App
+
+In your terminal, change directories to one where you wish to initialize a new Holochain app.
+Run the following, replacing `your_app_name` with your actual app name:
+```shell
+$ hcdev generate your_app_name
+$ cd your_app_name
+```
+
+We now have the empty shell of a Holochain app. From here, we will want to generate at least one Zome.
+To do this, run the following, replacing `your_zome_name` with a name related to the functionality you wish to develop. For example: `users`.
+```shell
+$ hcdev generate zomes/your_zome_name rust
+```
+
+Currently, Rust is the only language that the tool will scaffold a Zome for. In the future, other languages will be included. In the command above, we declared `rust` just to be explicit, even though it's the default language.
+
+What this did is generate a new folder under `zomes` called `users`. Here is the folder structure of it.
+- users
+  - code
+    - src
+      - main.rs
+    - .build
+    - Cargo.toml
+  - zome.json
+
+So in a given Zome we have two things:
+1. a JSON file, `zome.json`, which defines and configures the Zome
+2. a `code` folder, which can be compiled into a single `WASM` file with the code for this Zome
+
+In order for Holochain to run your app, you have to build your code into a single packaged file. Those instructions follow.
+
+## What are .hcpkg files?
+
+A Holochain app can be fully contained in a file known as a `.hcpkg` file.
+It is a JSON file, with a particular structure that Holochain can understand, and execute.
+
+This is an unusual JSON file, it is part configuration, and part executable.
+
+The configuration part comes from the `json` files that are throughout your app. One at the top level for the application (`app.json`) and one for each Zome. Ultimately, these get stitched together into a single tree structure in the `.hcpkg` file.
+
+The executable part comes from having embedded Base64 encoded WebAssembly code in the file. *What does that mean?* [WebAssembly](https://webassembly.org/) is a fast and secure low-level language.
+Rather than storing the code in its raw format, Holochain expects the code to be [encoded using Base64](https://en.wikipedia.org/wiki/Base64) , so that its as compact as possible.
+
+If you haven't heard of WebAssembly (WASM for short), that's ok. Important to know is that WASM is intended as a "compilation target" for other languages, not a language to write code in. So instead of writing code in WASM, write code in a language that's familiar to you, and [supports WASM](https://github.com/appcypher/awesome-wasm-langs). When it's time to run your code in Holochain, compile it.
+
+In order to avoid having to handcraft this complex JSON structure, with lots of room for error, the `hcdev package` command streamlines the process of taking your "raw" application folder, and packaging it up into the final `.hcpkg` file. 
```

Comment:
Yeah, I like introducing our own filename suffix. Did we use .hcpkg before in the Go version or did you just make that up? Anyway, I like it :)

Reply:
I didn't make it up, I saw it in use around so I thought it was settled lol

Reply:
yes i vaguely remember seeing this somewhere